packages/core/src/match.ts

```diff
@@ -69,11 +69,8 @@ function matchesReferenceFilter(resource: Resource, filter: Filter, searchParam:
       // We need to maintain this behavior for backwards compatibility
       match = references.some((reference) => reference?.endsWith('/' + filterValue));
     }
-    if (match && !negated) {
-      return true;
-    }
-    if (match && negated) {
-      return false;
+    if (match) {
+      return !negated;
```

Comment:
This is logically equivalent -- I'm not sure why I didn't write it this way in the first place.